### PR TITLE
ENH: update to pcds-5.0.0

### LINKED ISSUE #195: Add pytest-repeat
I'm enjoying this pytest extension offline for helping me reproduce/rule out race conditions.

It simply adds a `--count` argument to repeat every test n times.

```
$ pytest -v --count 1000 tests/test_positioner.py
```

### LINKED ISSUE #196: Add pytest-xdist
Useful plugin for running your tests on different hosts or different cores

Will be used in the gateway config test suite, should make it into the standard install so that anyone can run those

### LINKED ISSUE #197: Add whatrecord (pypi)
Ken's neat project

Add to pip-packages.txt

### PR DESCRIPTION
**Pull request by ZLLentz**
Auto-generated release notes!

Added the Following Packages
----------------------------

- aiohttp
- apischema
- async-timeout
- execnet
- graphql-core
- inflection
- jbig
- libzlib
- multidict
- platformdirs
- pytest-aiohttp
- pytest-forked
- pytest-repeat
- pytest-xdist
- typing-extensions
- whatrecord
- yarl

Removed the Following Packages
------------------------------

- jdcal

PCDS Package Updates
--------------------

|   Package    |  Old   |  New   |                        Release_Notes                         |
|--------------|--------|--------|--------------------------------------------------------------|
|    blark     | 0.1.6  | 0.1.7  |     https://github.com/pcdshub/klauer/releases/tag/v0.1.7     |
|    happi     | 1.9.0  | 1.10.0 |    https://github.com/pcdshub/happi/releases/tag/v1.10.0     |
| hutch-python | 1.11.2 | 1.12.0 | https://github.com/pcdshub/hutch-python/releases/tag/v1.12.0 |
|    hxrsnd    | 0.2.4  | 0.2.6  |    https://github.com/pcdshub/hxrsnd/releases/tag/v0.2.6     |
|     nabs     | 1.1.2  | 1.1.3  |     https://github.com/pcdshub/nabs/releases/tag/v1.1.3      |
| pcdsdevices  | 4.7.1  | 4.8.0  |  https://github.com/pcdshub/pcdsdevices/releases/tag/v4.8.0  |
| transfocate  | 0.5.0  | 0.5.2  |  https://github.com/pcdshub/transfocate/releases/tag/v0.5.2  |

Lab Community Package Updates
-----------------------------

| Package |  Old  |  New  |
|---------|-------|-------|
| bluesky | 1.7.0 | 1.8.0 |
|  ophyd  | 1.6.1 | 1.6.2 |

Python Community Core Package Updates
-------------------------------------

|   Package    |  Old   |  New   |
|--------------|--------|--------|
|   ipython    | 7.26.0 | 7.28.0 |
| scikit-learn | 0.24.2 |  1.0   |
|    sphinx    | 4.1.2  | 4.2.0  |

Other Python Community Major Updates
------------------------------------

|     Package      |  Old   |  New   |
|------------------|--------|--------|
|   cloudpickle    | 1.6.0  | 2.0.0  |
|    decorator     | 5.0.9  | 4.4.2  |
|        gh        | 1.14.0 | 2.0.0  |
|    jsonschema    | 3.2.0  | 4.0.1  |
|  jupyter_client  | 6.1.12 | 7.0.5  |
|      pluggy      | 0.13.1 | 1.0.0  |
|    setuptools    | 49.6.0 | 58.0.4 |
| sphinx_rtd_theme | 0.5.2  | 1.0.0  |
|  threadpoolctl   | 2.2.0  | 3.0.0  |
|      tzdata      | 2021a  | 2021b  |
|     tzlocal      |  2.1   |  3.0   |

To-do:
- [x] Add ~whatrecord~, pytest-xdist, pytest-repeat
- [x] Restore transfocate's keep-updated status
- [x] Add whatrecord tests
- [x] Update the tag specs
- [x] Switch the "next env" build to not allowed to fail (previously, was allowed to fail because otherwise it prevented the build from fully running, no longer the case- now I want to be notified when the next tag env will break with the travis cron build)
- [x] Merge `transfocate` conda-forge PR
- [x] Wait for the package to be available
- [x] Build new environment spec (should fail right now)
- [x] Push back to this branch and pray for magic
- [x] Fix issues with the first build
- [x] Get whatrecord back in, either by fixing the pypi build or by doing a pcds-tag conda build
- [x] Make auto release notes script

Related issues:
closes #195 
closes #196 
closes #197

### CODE REVIEW

**Comment by ZLLentz** [comment]
For a quick check of everything else, I'm going to disable whatrecord, which is failing to install, and see if all the other tests pass.

**Comment by klauer** [comment]
`whatrecord>=v0.1.2` is my latest attempt at getting the pip install to work. 🤞

File: envs/pcds/pip-packages.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 epicscorelibs==7.0.4.99.1.2
```
Comment (klauer) [inline envs/pcds/pip-packages.txt:1]:
```suggestion
```

Comment (ZLLentz) [reply envs/pcds/pip-packages.txt:1]: agh

**Comment by ZLLentz** [comment]
I need to force travis to run

**Comment by klauer** [comment]
🥳 🎆

File: scripts/release_notes_table.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    'jupyter',
+    'numpy',
+    'opencv',
+    'pandas',
+    'pytest',
+    'scikit-image',
+    'scikit-learn',
+    'scipy',
+    'sphinx',
+    'xarray',
```
Comment (ZLLentz) [inline scripts/release_notes_table.py:85]: @klauer any thoughts on packages that should be added/removed from this list? The consequence is these show for Major and Minor updates, while packages outside of the lists only show for Major updates.

Comment (klauer) [reply scripts/release_notes_table.py:85]: How about:
* `periodictable`  (or maybe that's a 'lab package' above?)
* `xraylib`

Maybe, maybe not:
* `h5py` (we hardly use it, but databroker experience makes me want to list it anyway)

Nothing else really jumps out - there are some that'd be useful for developers like us (`pytest*`).

**Review by klauer — APPROVED** [review]
✅ New packages look good (thanks!)
✅ Auto-generated diff looks great
✅ Tests are passing

All LGTM. `5.0.0`!

File: scripts/release_notes_table.py (hunk 2 above)
Comment (ZLLentz) [reply scripts/release_notes_table.py:85]: Those are all reasonable, I'll add them all

**Comment by ZLLentz** [comment]
ok I think it is time to lock this in and merge

**Comment by ZLLentz** [comment]
waiting for travis since I won't deploy until the PAMM tomorrow

**Pull request merged**